### PR TITLE
CVSL-541: Update probation team on licence when offender manager is updated

### PR DESCRIPTION
**Pull request by smcveigh941**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt [hunk 1]
```diff
@@ -62,4 +64,45 @@ class OffenderController(private val offenderService: OffenderService, private v
...
+      )
+    ]
+  )
+  @Transactional
+  fun updateProbationTeam(
+    @PathVariable crn: String,
+    @RequestBody body: UpdateProbationTeamRequest
+  ) {
+    this.offenderService.updateProbationRegion(crn, body)
+  }
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt:107]: Just naming potentially? `updateProbationTeam` calling a service method called `updateProbationRegion`? Probation region is a specific value. Small one, not a blocker.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt [hunk 2]
```diff
@@ -39,4 +40,47 @@ OffenderService(
       )
     }
   }
+
+  @Transactional
+  fun updateProbationRegion(crn: String, request: UpdateProbationTeamRequest) {
+    val inFlightLicenceStatuses = listOf(LicenceStatus.IN_PROGRESS, LicenceStatus.SUBMITTED, LicenceStatus.APPROVED, LicenceStatus.ACTIVE)
+    var offenderLicences = this.licenceRepository.findAllByCrnAndStatusCodeIn(crn, inFlightLicenceStatuses)
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt:47]: Should we include VARIATION_IN_PROGRESS, VARIATION_SUBMITTED and VARIATION_APPROVED / REJECTED too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt (hunk 1 above)
Comment (smcveigh941) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderController.kt:107]: Sorted now, thanks

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt (hunk 2 above)
Comment (smcveigh941) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt:64]: Oh! Yes we should, looks like we missed it on updating the COM too. Fixed now

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**